src/backup.rs

```diff
@@ -0,0 +1,107 @@
+//! Backup Power Domain
```

Comment:
I think it makes sense to move this file into the rcc folder. You can then have `pub use` statements in rcc/mod.rs to export `Backup` and (conditionally) `rtc`.